### PR TITLE
DataAccess is now correctly created.

### PR DESCRIPTION
**Pull request by gumbarros**
@LucioPelinson .

Currently `DataAccess` is "chumbado" with a ConnectionString named `ConnectionString`.

This solves the problem:
```cs
    public JJServiceBuilder WithDataAccess(Func<IServiceProvider, DataAccess> implementationFactory)
    {
        Services.Replace(ServiceDescriptor.Scoped(implementationFactory));
        return this;
    }
```

This will be merged at `main`, we're needing the fix.

### CODE REVIEW

File: src/JJMasterData.Commons/Data/DataAccess.cs [hunk 1]
```diff
@@ -97,6 +99,7 @@ public DbProviderFactory Factory
     /// <summary>
     /// By default DataAccess recover connection string from appsettings.json with name ConnectionString
     /// </summary>
+    [Obsolete("Will be removed at Sun version, this constructor uses a static service locator and can cause runtime errors.")]
```
Comment (LucioPelinson) [inline src/JJMasterData.Commons/Data/DataAccess.cs]: We can't force it that way, nothing prevents using it without dependency injection

**Review by LucioPelinson — CHANGES_REQUESTED** [review]

File: src/JJMasterData.Commons/Data/DataAccess.cs (hunk 1 above)
Comment (gumbarros) [reply src/JJMasterData.Commons/Data/DataAccess.cs]: I strongly disagree, this constructor is impossible to exist at .NET Core without the static service locator (`JJService`) and causes a lot of headache when people forget to call `UseJJMasterData` (that I also want to remove this additonal step at Sun) at `Program.cs` .

If you don't want to use dependency injection, this constructor is perfectly fine :
```cs
public DataAccess(string connectionString, string connectionProvider)
{
///....
}
```

If you really want that empty constructor, we can write a #ifdef to only work at .NET Framework, but in .NET Core it causes lots of problems.

File: src/JJMasterData.Commons/DI/JJServiceBuilder.cs [hunk 2]
```diff
@@ -50,7 +53,13 @@ public JJServiceBuilder AddDefaultServices(IConfiguration configuration)
         Services.Replace(ServiceDescriptor.Transient<IBackgroundTask, T>());
         return this;
     }
-
+    
+    public JJServiceBuilder WithDataAccess(Func<IServiceProvider, DataAccess> implementationFactory)
```
Comment (LucioPelinson) [inline src/JJMasterData.Commons/DI/JJServiceBuilder.cs]: I think nobody needs to know about the existence of the DataAccess class

Comment (gumbarros) [reply src/JJMasterData.Commons/DI/JJServiceBuilder.cs]: Then how you customize the connection string that you want MasterData to use? (@ivanilson22  problem)

**Pull request merged**